src/audio/copier/copier.c

```diff
@@ -295,6 +296,95 @@ static enum sof_ipc_stream_direction
 	}
 }
 
+static int init_dai_single(struct comp_dev *parent_dev,
+			   const struct comp_driver *drv,
+			   struct comp_ipc_config *config,
+			   const struct ipc4_copier_module_cfg *copier,
+			   struct pipeline *pipeline,
+			   struct ipc_config_dai *dai,
+			   enum ipc4_gateway_type type,
+			   int index)
+{
```

Comment:
to really review this I'd need to compare it to `init_dai()` below... And I think they have a lot of common code. Wouldn't the change both look smaller and be easier to review if we kept a single function? Possibly similar changes are made and could be simplified in other commits in this PR too

Reply:
@lyakh it looks like it in this commit but look at the same function after the last commit and you'll see the difference. It just makes it a lot easier this way.

Reply:
@ranj063 sorry, you review each patch. You don't just review the result.

Reply:
@lyakh i've split up the patches now. The number of patches are twice as much now. But hopefully each one is small enough to make it easier to review.